sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java

```diff
@@ -358,7 +386,14 @@ public void skipIntegers(int total) {
     while (left > 0) {
       if (this.currentCount == 0) this.readNextGroup();
       int n = Math.min(left, this.currentCount);
-      advance(n);
+      switch (mode) {
+        case RLE:
+          break;
+        case PACKED:
+          currentBufferIdx += n;
+          break;
+      }
+      currentCount -= n;
```

Comment:
Hmm, this is a old bug?

Reply:
no - it used to call `advance` which does the same thing, but since we removed the method I just replaced it with the old method body

Reply:
Do you remove `advance()` method? I don't see it is removed here.

Reply:
hmm somehow it survived - let me remove it for real.